src/main/java/gregtech/api/recipes/map/MapItemStackNBTIngredient.java

```diff
@@ -54,35 +47,20 @@ public boolean equals(Object obj) {
             if (this.meta != other.meta) {
                 return false;
             }
-            if (this.matcher != null && other.matcher != null) {
-                if (!this.matcher.equals(other.matcher)) {
-                    return false;
+            if (this.gtRecipeInput != null) {
+                if (other.gtRecipeInput != null) {
+                    return gtRecipeInput.equals(other.gtRecipeInput);
                 }
+            } else if (other.gtRecipeInput != null) {
+                return other.gtRecipeInput.acceptsStack(this.stack);
```

Comment:
Now this function is near copy of function with same name in `MapItemStackIngredient` would it be possible to extract the same part and use it in both?